### PR TITLE
update next pages quickstart to revise todo state and authenticator placement

### PR DESCRIPTION
**Pull request by Jay2113**
#### Description of changes:

There is an issue with the current quickstart where due to the placement of the Authenticator component, a second user could sign in to the application and view the first user's todo list.

This PR updates next pages quickstart to place the Authenticator higher in the tree, re-rendering the App component and starting from empty state.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx [hunk 1]
```diff
@@ -221,39 +221,28 @@ This should start a local dev server at http://localhost:3000.
...
-
-```tsx title="pages/index.tsx"
+```tsx title="pages/_app.tsx"
 // highlight-start
 import { Authenticator } from '@aws-amplify/ui-react'
 import '@aws-amplify/ui-react/styles.css'
 // highlight-end
-// ... other imports
+import "@/styles/app.css";
+import type { AppProps } from "next/app";
```
Comment (josefaidt) [inline src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx:232]: import sorting nit -- can we put the type import at the top of the file?

File: src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx [hunk 2]
```diff
@@ -378,24 +396,23 @@ export const data = defineData({
...
 
-In the application client code, let's also render the username to distinguish different users once they're logged in. Go to your **src/App.tsx** file and render the `user` property.
+In the application client code, let's also render the username to distinguish different users once they're logged in. Go to your **pages/index.tsx** file and render the `user` property from the `useAuthenticator` hook.
 
 ```tsx title="pages/index.tsx"
 // ... imports
 
 function App() {
+  // highlight-next-line
+  const user = useAuthenticator().user;
```
Comment (josefaidt) [inline src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx:406]:
```suggestion
  const { user, signOut } = useAuthenticator();
```

File: src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx [hunk 3]
```diff
@@ -305,6 +294,35 @@ export const auth = defineAuth({
...
+```tsx title="pages/index.tsx"
+import type { Schema } from "@/amplify/data/resource";
+// highlight-next-line
+import { useAuthenticator } from "@aws-amplify/ui-react";
+import { useState, useEffect } from "react";
+import { generateClient } from "aws-amplify/data";
+
+const client = generateClient<Schema>();
+
+export default function App() {
```
Comment (josefaidt) [inline src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx:308]:
```suggestion
export default function HomePage() {
```

File: src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx [hunk 4]
```diff
@@ -378,24 +396,23 @@ export const data = defineData({
 });
 ```
 
-In the application client code, let's also render the username to distinguish different users once they're logged in. Go to your **src/App.tsx** file and render the `user` property.
+In the application client code, let's also render the username to distinguish different users once they're logged in. Go to your **pages/index.tsx** file and render the `user` property from the `useAuthenticator` hook.
 
 ```tsx title="pages/index.tsx"
 // ... imports
 
 function App() {
```
Comment (josefaidt) [inline src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx:404]:
```suggestion
function HomePage() {
```

File: src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx [hunk 5]
```diff
@@ -378,24 +396,23 @@ export const data = defineData({
...
-      {({ signOut, user }) => (
-        <main>
-          // highlight-next-line
-          <h1>{user?.signInDetails?.loginId}'s todos</h1>
-          {/* ... rest of the UI */}
-        </main>
-      )}
-    </Authenticator>
+      <h1>{user?.signInDetails?.loginId}'s todos</h1>
+      {/* ... rest of the UI */}
```
Comment (josefaidt) [inline src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx:414]:
```suggestion
      {/* ... */}
```

nit to align with other comments

File: src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx (hunk 2 above)
Comment (Jay2113) [reply src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx]: Done

File: src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx (hunk 3 above)
Comment (Jay2113) [reply src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx]: Done

File: src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx (hunk 4 above)
Comment (Jay2113) [reply src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx]: Done

File: src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx (hunk 5 above)
Comment (Jay2113) [reply src/pages/[platform]/start/quickstart/nextjs-pages-router/index.mdx]: Done

**Review by josefaidt — APPROVED** [review]

**Review by cwomack — APPROVED** [review]

**Review by esauerbo — APPROVED** [review]

**Pull request merged**